### PR TITLE
Add several software topics

### PR DESCRIPTION
**Pull request by bkreynen**
Adds several software related topics and includes a small colour change to the website to make it fit a bit more with QuTech.

### CODE REVIEW

File: docs/software/FAIR.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# FAIR Software (QSSM 2)
```
Comment (QFer) [inline docs/software/FAIR.md]: QSSM -> QSMM (stands for QuTech Software Maturity Model)

File: docs/software/FAIR.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+A lot of research software also falls in category 2. With this software it is important that it can be reproduced. Whether that is by yourself when you want to run a similar or the same experiment again, or by someone else. Software that is used to actually run an experiment or analysis of data used in papers falls in this category. We often see that software that should have this maturity level is not actually easy to reproduce within QuTech. While making it reproducible often requires you to invest a little bit more time in the beginning of your project, we expect that throughout the duration of the project it will end up saving your time. We also provide you with some resources in the next section to get you started quickly.
+
+## What?
+
+* Most software at QuTech is developed in Python. We have a template to get you started quickly with these principles in Python. With the template you can quickly create your python project adhering to QSSM 2 by answerring a few simple questions. The template can be foud on [github](https://github.com/QuTech-Delft/FAIR-Python-script-template).
```
Comment (QFer) [inline docs/software/FAIR.md]: QSSM 2 -> QSMM level 2

Comment (QFer) [inline docs/software/FAIR.md]: answerring -> answering

Comment (QFer) [reply docs/software/FAIR.md]: foud -> found

File: docs/software/FAIR.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  * (1.5 GS credits) A [course about Git](https://www.tudelft.nl/library/research-data-management/r/training-evenementen/training-voor-onderzoekers/version-control-collaborative-development-for-research-software)
+  * (2.5 GS credits) A [more general course](https://www.tudelft.nl/library/research-data-management/r/training-evenementen/training-voor-onderzoekers/coderefinery-workshops) to refine your coding skills, which includes an introduction to Git
+  * (1.5 GS credits) A [programming introduction course](https://www.tudelft.nl/library/research-data-management/r/training-evenementen/training-voor-onderzoekers/software-carpentry-workshops) for researches with little to no experience with programming, which includes an introduction to Git
+* For those that are interested in software quality in general other pages on this site also provides some info on other related topics and additional resources to learn more.
+* The SDST is also available if you have questions about any of these topics or if need some help with using the templates for example.
```
Comment (QFer) [inline docs/software/FAIR.md]: or if need -> or if you need

File: mkdocs.yml [hunk 4]
```diff
@@ -10,9 +10,15 @@ nav:
       - data/index.md
     - Software: 
       - software/index.md
+      - "FAIR Software (QSSM 2)": software/FAIR.md
```
Comment (QFer) [inline mkdocs.yml]: QSMM

File: docs/software/testing.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+Software testing has the following advantages:
+
+* It improves the maintainability of your code. By automatically being able to test your software you can be more certain that you did not break anything in your code.
+* It makes it easier to spot mistakes. Having automated tests can help you pinpoint where in your software you have made a mistake.
+* It gives you, and others, more confidence in the correctness of the results.
+* Since you define the expected inputs and outputs tests can also serve as a form of documentation. It makes clear what the expectations are for a certain piece of your software and therefore also gives some information on how it can be used.
+
+## What?
+
+When we talk about (automated) software testing then we define a few types of tests the main categories are:
```
Comment (QFer) [inline docs/software/testing.md]: ...tests. The main...

File: docs/software/testing.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+* End-to-end tests: These test the full working of your software. In the case of a simple script this could be a test which provides an input file and checks if the expected output file is created.
+* Unit tests: Unit tests test a smaller piece (unit) of your software. This can be for example a test of a single function which is used to process your results.
+* Integration tests:   Integration tests test tests your integrations with 3rd party services. E.g. if your application makes use of some third party software you test if the system works as expected when you combine your code with this third party software.
```
Comment (QFer) [inline docs/software/testing.md]: 1 tests too many

File: docs/software/testing.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+To start with software testing you need to consider two important questions:
+
+* What is the expected behaviour of the software on certain inputs?
+* What are the edge cases of this behaviour?
+  * With this we mean: for which cases should the software do something differently than the “normal” flow. A simple example is a division function. If the divider is 0 then a division is not possible so instead an error should be thrown. In our tests we would therefore check:
+    * That the correct result is produced when the divider is something different than 0.
+    * That an error is thrown when the divider is 0.
+
+If you have answered these two questions you can start creating test cases which check this behaviour and their edge cases.
+Courses
```
Comment (QFer) [inline docs/software/testing.md]: Is this a separate paragraph or chapter?

File: docs/software/testing.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+Want to learn more about software testing?
+
+* If you want to increase your software skills in general then the library offers a [coderefinery course](https://www.tudelft.nl/library/research-data-management/r/training-evenementen/training-voor-onderzoekers/coderefinery-workshops) which you can get graduate school credits for. This course is meant for people who already are able to program but would like to improve their skills. It also includes a part about software testing.
```
Comment (QFer) [inline docs/software/testing.md]: software skills -> software development skills

File: docs/software/vcs.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
+# Version Control Systems
+This page gives some information about version control systems. If you are already familiar with those you might want to skip ahead to see if the additional resources offer something new to learn for you. On this page we give a short description of the version control infrastructure available to researches at QuTech. If version control systems are new for you then will begin by explaining which problems they can help solve, how they do this and what tools you can use for this.
```
Comment (QFer) [inline docs/software/vcs.md]: "then will begin by" is a bit unclear what you mean "then we will begin by"?.

File: docs/software/vcs.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+Additionally version control systems have a concept called branches, which allows for more advanced collaboration. If you just want to use Git as a solo developer you probably do not need to think about this. We briefly cover the topic of branches here though: Branches allow you to “branch off”. When you do this you create a new copy of your code which is stored separately from your main code (stored on the main branch). You can then start making changes on this branch and once you are finished you can add the changes you have made back your main branch, incorporating them into your “normal” code. This allows for more collaboration on the same codebase. Different features for example can be worked on simultaneously on different branches.
```
Comment (QFer) [inline docs/software/vcs.md]: Here sudden;ly Git is introduced. I think it is better to stick with vcs here.

File: docs/software/vcs.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+There are different philosophies about how you can use these branches to collaborate, each having their own advantages and disadvantages. We will give some more information on these different workflows in the additional resources. If you are unsure how best work with the branches for your project you can also ask the SDST for advice.
+
+## What?
+
+There are different version control systems. At QuTech mostly Git is used, this is also one of the most popular version control systems. The additional resources section provides information on learning Git if you are not yet familiar with it. If you are familiar it also offers some advanced topics if you want to learn more.
```
Comment (QFer) [inline docs/software/vcs.md]: Maybe bold face Git

File: docs/software/vcs.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+As mentioned version control systems store your code remotely. This means that next to a system such as git you also need a server to store the code on. There are a few options for this. The two most popular options are GitHub and GitLab. At QuTech we also have the possibility to use a version of GitLab hosted by the TU Delft. Your research group might already have a preferred option so you might want to ask your supervisor what is used in your group. If your research group does not have a preferred option yet you can ask the SDST for help with selecting one.
```
Comment (QFer) [inline docs/software/vcs.md]: git -> Git

File: docs/software/vcs.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+  * [Fork](https://git-fork.com/)
+    * Free at the moment, most likely will cost some money in the future
+  * [GitHub Desktop](https://desktop.github.com/)
+    * Free. Generally easy to use, but tailored to GitHub specifically
+
+#### Beginners
+
+Want to learn the basics of Git?
+
+* The TU Delft library offers three different courses which teach you about git (and give you graduate school credits!)
```
Comment (QFer) [inline docs/software/vcs.md]: git -> Git

**Review by QFer — CHANGES_REQUESTED** [review]
Nice piece of work!

File: docs/software/testing.md (hunk 7 above)
Comment (bkreynen) [reply docs/software/testing.md]: Yes it should have been, was not copied over correctly from confluence.

**Review by QFer — APPROVED** [review]
